cidr.go

```diff
@@ -188,6 +194,27 @@ func (c CIDR) SubNetting(method SubNettingMethod, num int) ([]*CIDR, error) {
 	return cidrArr, nil
 }
 
+// SubNet split network segment based on a new subnet prefix
+// If split via prefix_diff, use Subnetting() instead
+func (c CIDR) SubNet(newPrefix int) ([]*CIDR, error) {
```

Comment:
Instead of adding a similar method, you can extend the method type 'MethodMaskPrefix' of the SubNetting()